### PR TITLE
Exams

### PR DESCRIPTION
**Pull request by GAlexandrD**

### CODE REVIEW

**Comment by GAlexandrD** [comment]
alternative implementation of distinct:
`const distinct = (array) => Array.from(new Set(array));`
it seems to me that plane can be optimized using closure:
```
const plane = (array) => {
  const result = [];
  const closedPlane = (array) => {
    for (const item of array) {
      if (Array.isArray(item)) closedPlane(item);
      else result.push(item);
    }
    return result;
  };
  return closedPlane(array);
};
```
We don't create temporary arrays and don't use unnecessary rest operators and push calls. But option with closure is larger.

**Comment by tshemsedinov** [comment]
- Prefer for..of
- Don't change incoming parameters

**Comment by tshemsedinov** [comment]
What is easy to read recursion with closure or just a recursion?

**Comment by GAlexandrD** [comment]
> * Prefer for..of
> * Don't change incoming parameters

i don't understand where i change parameter? I copied it in the result.

File: Tasks/distinct.js [hunk 1]
```diff
@@ -1,16 +1,14 @@
...
-        if (A.has(a)) {
-        data.splice(w, 1)
-        } else {
-        A.add(a)
-        };
-        w++;
-    });
-  return data;}
+const distinct = (array) => {
+  const result = [...array];
```
Comment (GAlexandrD) [inline Tasks/distinct.js:4]: > * Prefer for..of
> * Don't change incoming parameters

I changed result, is it parameter? this way create new array and copy each element, I tried array === result and it showed false.

**Pull request closed without merging**